### DISCUSSION TITLE
Getting rid of group.node_data

### DISCUSSION THREAD

**Discussion by ipspace** [ipspace/netlab#596] [Features]
Now that we'll finally handle attributes of ~~all~~ most objects in netlab topology, we could get rid of **group.node_data** and just define node attributes within a group dictionary.

The attributes would be validated using a combination of `[ group, node ]` attributes + providers (so we would know they're valid) and everything but group attributes would be copied into node data.

As a transition path, the early group data structure validation would copy everything in `node_data` back into the group dictionary and pop `node_data` dictionary. After a long while, we would make `node_data` attribute invalid.

Thoughts @ssasso @jbemmel?

**Comment by jbemmel** [comment]
Makes sense, should be a low impact change. Probably want to include a deprecation warning for node_data

**Comment by ipspace** [comment]
Implemented in b78ca8098b63c88eb57ee9341612180552016632
